### ISSUE TITLE
Cannot read property 'translate' of undefined in karma spec test

### ISSUE THREAD

**Issue by Siedlerchr** [ngneat/transloco#250]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request
[ ] Other... Please describe:
</code></pre>

## Current behavior

Spec test fails and says `translate(....)`is undefined.

## Expected behavior

Spec test runs sucessful and translate does not throw an error.

## Minimal reproduction of the problem with instructions

Create a component and call  `translate('some key')` somewhere in it.
`import { translate } from '@ngneat/transloco';`

Create a spec test (default generated)
Add the following import:
import { getTranslocoModule } from '../transloco-testing-module';

```
describe('MapContainerComponent', () => {
  let component: MapContainerComponent;
  let fixture: ComponentFixture<MapContainerComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ MapContainerComponent ],
      imports: [HttpClientTestingModule, RouterTestingModule, getTranslocoModule()],
      providers: [Network],
      schemas: [CUSTOM_ELEMENTS_SCHEMA],
    })
    .compileComponents();
  }));
```

Transloco Testing Module:
```
import { TranslocoTestingModule, TranslocoConfig } from '@ngneat/transloco';
import en from '../assets/i18n/en-US.json'
import de from '../assets/i18n/de-DE.json'

export function getTranslocoModule(config: Partial<TranslocoConfig> = {}) {
  return TranslocoTestingModule.withLangs(
    { en, de },
    {
      availableLangs: ['en-US', 'de-DE'],
      defaultLang: 'en-US',
      reRenderOnLangChange: true,
      ...config
    }
  );
}
```

## Environment

<pre><code>
Angular version: 8.1.2
```
    "@ngneat/transloco": "^2.13.5",
    "@ngneat/transloco-locale": "^1.0.1",
```

Browser:
- [x] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [x] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: 12.14.1
- Platform:  Windows, Linux

Others:

</code></pre>

**Comment by stijnvn** [comment]
I can confirm this bug also exists in combination with Angular 9.0.2.

**Comment by ArielGueta** [comment]
Works for me. Can you reproduce it?

**Comment by shaharkazaz** [comment]
@Siedlerchr  Feel free to reopen the issue and provide a reproduction 🙂 .

**Comment by tommueller** [comment]
I actually just ran into the same issue and solved it by importing my `TranslocoRootModule` additionally to the `TranslocoTestingModule`. So basically like this:

```typescript
const createComponent = createComponentFactory({
  component: AComponent,
  imports: [
    TranslocoRootModule, // not working without this!
    getTranslocoTestingModule()
  ]
});
```

with the component looking like this:

```typescript
import { Component, Input } from '@angular/core';
import { translate } from '@ngneat/transloco';

@Component({...})
export class ActionWrapperComponent {
  @Input() label = translate('ACTIONS.EDIT');
```

**Comment by tommueller** [comment]
Nope my fix did not work. I still get the error, but only sometimes.
Around 30% of my test-runs fail with this bug. Will try to produce a repro when I find the time.

**Comment by tommueller** [comment]
@shaharkazaz here is a repro: https://github.com/tommueller/translate-repro

**Comment by shaharkazaz** [comment]
@tommueller I'll take a look when I get the chance 🙂

**Comment by DaveGold** [comment]
I also experience an issue when using the getTranslocoModule.
I am testing some code that is using translocoService.translate but the key is not translated in the unit test but it does in the application.

**Comment by shaharkazaz** [comment]
@DaveGold are you sure it's related to this issue?

**Comment by developer239** [comment]
Why is this closed? 😄

**Comment by shaharkazaz** [comment]
@developer239 I personally didn't have time to get to it in the last couple of months so it just got missed, thanks for bringing this up again 🙂 

@tommueller Thank you for the reproduction, here is a summary of the issues discussed above:
There are 2 issues here that got mixed together:
1. `TypeError: Cannot read property 'translate' of undefined` thrown when using the pure translate function.
2. Values don't get translated when using the pure `translate` or the `translocoService.translate`.

### Explnations

1. This issue was caused since the pure `translate` function is a proxy the `translocoService.translate` so while in your actual app the service was already initialized when you got to that component if you didn't provide it as part of your testing you get `TypeError: Cannot read property 'translate' of undefined`.

2. This is [stated in the docs](https://ngneat.github.io/transloco/docs/translation-api#translate), and also applies to the specs, it's your responsibility to make sure that the translations are already loaded when using the synchronous translate function.

### Solutions
1. I'll update the testing module to take care of the injection.
2. While this behavior makes sense in the real app, it could be tiring to make sure the translations are loaded in specs, I'll add an option to preload all the given langs via the module. (`{ preloadLangs: true }`).
You only need to use this option if you aren't using transloco in the template since if you are it takes care of the loading for you.
Note that as part of this change `TranslocoTestingModule.withLangs` is deprecated (still supported and also supports the new option but will be removed in v3) and you should now use `TranslocoTestingModule.withOptions`. 
3. I'll update all the docs accordingly.

**Comment by tommueller** [comment]
Thanks, will give feedback ASAP once it's released.

**Comment by shaharkazaz** [comment]
@tommueller released in v2.20.0

**Comment by tommueller** [comment]
@shaharkazaz sorry for the late response, I was able to enjoy some extend holiday season.

Just checked out the v2.20.0 release and can confirm that the change fixes the issue. Thanks a bunch!
